### PR TITLE
p2p: minor cleanup + update router options

### PR DESCRIPTION
**Pull request by alexanderbez**
I wanted to address the point `Merge MaxNumInboundPeers and MaxNumOutboundPeers into MaxConnections.` from the p2p tracking issue, but realized we cannot remove those fields until we remove _all_ of the legacy stack.

So this PR just tweaks the current router options to use both max in and out plus some minor cleanup:

- Rename legacy flag
- Add new `MaxIncomingConnectionAttempts` config
- Add new `MaxConnections` config (will replace `MaxNumInboundPeers` and `MaxNumOutboundPeers`)

/cc @tessr

### CODE REVIEW

File: test/e2e/networks/ci.toml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # This testnet is run by CI, and attempts to cover a broad range of
 # functionality with a single network.
 
-initial_height = 1000
+disable_legacy_p2p = false
```
Comment (alexanderbez) [inline test/e2e/networks/ci.toml:4]: You can ignore most of these changes. My vsc TOML extension automatically sorts :)

Comment (tessr) [reply test/e2e/networks/ci.toml:4]: maybe we should all be running that extension, tbh

Comment (alexanderbez) [reply test/e2e/networks/ci.toml:4]: yah I think we should ^__^

File: config/config.go [hunk 2]
```diff
@@ -567,11 +567,25 @@ type P2PConfig struct { //nolint: maligned
...
 	MaxNumOutboundPeers int `mapstructure:"max-num-outbound-peers"`
 
+	// MaxConnections defines the maximum number of connected peers (inbound and
+	// outbound).
+	MaxConnections uint16 `mapstructure:"max-connections"`
+
+	// MaxIncomingConnectionAttempts rate limits the number of incoming connection
+	// attempts per IP address.
+	MaxIncomingConnectionAttempts uint `mapstructure:"max-incoming-connection-attempts"`
+
```
Comment (cmwaters) [inline config/config.go:588]: Can we add these to the `DefaultConfig`/ `DefaultP2PConfig` funcs

**Comment by alexanderbez** [comment]
Good question @cmwaters -- I'm not sure tbh.

File: config/config.go (hunk 2 above)
Comment (alexanderbez) [reply config/config.go:588]: Good call. Done :)

**Review by tychoish — APPROVED** [review]

**Pull request merged**